### PR TITLE
Move validation builds to official ado agents

### PR DESCRIPTION
**Pull request by arroyc**
We will have multiple PRs  to get rid of oryxlinux pool .... first phase was base image builds, detector builds and platform binaries builds. We already have done that, this is our second phase. Next will be nightly builds and at the end we will move ci build.

We are taking phase wise approach as we dont want to block ci/cd for this

### CODE REVIEW

File: vsts/pipelines/templates/_buildTemplate.yml [hunk 1]
```diff
@@ -29,6 +29,14 @@ steps:
     startsWith(variables['Build.SourceBranch'],'refs/heads/patch/'),
     startsWith(variables['Build.SourceBranch'],'refs/heads/exp/')))
 
+- script: |
+    sudo rm -rf /usr/share/dotnet
+    sudo rm -rf /opt/ghc
+    sudo rm -rf "/usr/local/share/boost"
+    sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+    docker images && docker system prune -fa && df -h && echo
+  displayName: 'clean docker images'
```
Comment (qianz2) [inline vsts/pipelines/templates/_buildTemplate.yml:38]: Will it be faster if not doing it. Sometimes if it's cached, then build build images step will be faster?

Comment (qianz2) [reply vsts/pipelines/templates/_buildTemplate.yml:38]: Such as triggering to re-run failed job directly from the pipeline, I don't know if that's an example

Comment (arroyc) [reply vsts/pipelines/templates/_buildTemplate.yml:38]: unfortunately we cant ... first we dont know which agent we are gonna get .. these are auto managed by ado team .. so they recycle these agents .. secondly there is space limitation .. only 10-12 gb available ..we are sacrificing build speed here over build security @qianz2

Comment (qianz2) [reply vsts/pipelines/templates/_buildTemplate.yml:38]: I see, thanks for clarifying, then I agree we should clean docker images.

**Review by qianz2 — APPROVED** [review]

**Pull request merged**